azure-client-runtime/src/main/java/com/microsoft/azure/v2/annotations/AzureHost.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.v2.annotations;
+
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.AzureEnvironment.Endpoint;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+
+@Target(value={METHOD,TYPE})    // The context in which annotation is applicable i.e. this annotation (Host) can be applied to method, method parameter and interface.
+@Retention(RetentionPolicy.RUNTIME)        // Record this annotation in the class file and make it available during runtime.
+public @interface AzureHost {
+    String value() default "";
+    AzureEnvironment.Endpoint enpoint() default Endpoint.RESOURCE_MANAGER;
+}
+
+/**
+ * The applicability of this annotation is limited to variable with name {host} in the URI template.
```

Comment:
These comments are great, well written, and thought-out. Why not make them Javadoc comments on the @interface type itself so that they're available while we're writing code?